dissect/target/plugins/os/unix/log/auth.py

```diff
@@ -266,47 +285,81 @@ def _create_event_descriptor(self, record_fields) -> TargetRecordDescriptor:
+            CentOS format: Jan 12 13:37:00 hostname daemon: message
+            Debian format: Jan 12 13:37:00 hostname daemon[pid]: pam_unix(daemon:session): message
+            Ubuntu  24.04: 2024-01-12T13:37:00.000000+02:00 hostname daemon[pid]: pam_unix(daemon:session): message
+
+        Resources:
+            - https://help.ubuntu.com/community/LinuxLogFiles
+        """
 
         tzinfo = self.target.datetime.tzinfo
 
         var_log = self.target.fs.path("/var/log")
         for auth_file in chain(var_log.glob("auth.log*"), var_log.glob("secure*")):
-            for idx, (ts, line) in enumerate(year_rollover_helper(auth_file, RE_TS, "%b %d %H:%M:%S", tzinfo)):
-                ts_and_hostname = re.search(RE_TS_AND_HOSTNAME, line)
-                if not ts_and_hostname:
-                    self.target.log.warning("No timestamp and hostname found on line %d for file %s.", idx, auth_file)
-                    self.target.log.debug("Skipping line %d: %s", idx, line)
-                    continue
-
-                info = line.replace(ts_and_hostname.group(0), "").strip()
-                service, _message = info.split(":", maxsplit=1)
-                message = _message.strip()
-                # Get the PID, if present. Example: CRON[1] --> pid=1
-                pid = None
-                if "[" in service:
-                    service, _pid = service.split("[")[:2]
-                    pid = _pid.strip("]")
-
-                yield self._auth_log_builder.build_record(ts, auth_file, service, pid, message)
+            if is_iso_fmt(auth_file):
+                iterable = iso_readlines(auth_file)
+
+            else:
+                iterable = year_rollover_helper(auth_file, RE_TS, "%b %d %H:%M:%S", tzinfo)
+
+            for ts, line in iterable:
+                yield self._auth_log_builder.build_record(ts, auth_file, line)
+
+
+def iso_readlines(file: TargetPath, limit: int | None = None) -> Iterator[tuple[datetime, str]]:
+    """Iterator reading the provided auth log file in ISO format. Mimics ``year_rollover_helper`` behaviour."""
+
+    with open_decompress(file, "rt") as fh:
+        for count, line in enumerate(fh.readlines()):
+            if limit and count > limit:
+                break
+
+            if not (match := RE_TS_ISO.match(line)):
+                log.warning("No timestamp found in one of the lines in %s!", file)
+                log.debug("Skipping line: %s", line)
+                continue
+
+            try:
+                ts = datetime.strptime(match[0], "%Y-%m-%dT%H:%M:%S.%f%z")
+
+            except ValueError as e:
+                log.warning("Unable to parse ISO timestamp in line: %s", line)
+                log.debug("", exc_info=e)
+                continue
+
+            yield ts, line
+
+
+def is_iso_fmt(file: TargetPath) -> bool:
+    """Determine if the provided auth log file uses new ISO format logging or not."""
+
+    results = list(iso_readlines(file, limit=3))
+    return any(results)
```

Comment:
```suggestion
    return any(iso_readlines(file, limit=3))
```
Makes it lazy and return on the first result, instead of completely exhausting the log file.